### DISCUSSION TITLE
Gain maps in JPEG XL files

### DISCUSSION THREAD

**Discussion by jonsneyers** [libjxl/libjxl#3505] [JPEG XL specification extensions]
Currently we don't define a way to embed gain maps in JPEG XL files. There is currently an ongoing standardization effort by ISO/TC 42 (ISO 21496-1, https://www.iso.org/standard/86775.html?browse=tc) that could also be relevant for JPEG XL.

JPEG XL supports HDR natively, so there is no real need to use gain maps to represent HDR image data (unlike formats that are limited to 8-bit precision). But inverse gain maps (i.e. describing a custom tone mapping) could still be useful.

This is also somewhat related to https://github.com/libjxl/libjxl/discussions/3504, in case the JPEG XL file represents a recompressed JPEG with a gain map.

This thread can be used to discuss what would be the best way to represent gain maps in JPEG XL files, what changes/additions to ISO/IEC 18181-2 (JPEG XL File Format) would be desired, and what changes/additions to libjxl would be desired.

**Comment by Qhilm** [comment]
Technically, I believe color information rather than simple brightness information in the inverse gain map ("tone mapping map"?) would be helpful.

From a more general point of view, whatever implementation choices that allow for having the most robust / safest SDR version of the JXL HDR image is what I would prefer. By "safest", I mean the one that is most likely to work on the largest proportion of devices using SDR screens.

Currently, the tone mapping of an HDR JXL image can produce significantly different results, depending on the software / algorithm used for tone mapping (at least it has been my experience). The lack of "safe" SDR version (or "base SDR image" as I have seen it referred to) is for my usage (recreational photography) the biggest weakness of JPEG XL at the moment, especially when compared to ultra HDR (jpeg with gain map).

**Comment by gregbenz** [comment]
@jonsneyers The JXL-specific spec is now final, right? Adobe software can export JXL gain maps and the Adobe Gain Map Demo app reports them as being compliant to ISO 21496-1.
